### PR TITLE
Remove FixedLengthBinary and VariableLengthBinary traits

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Remove `FixedLengthBinary` and `VariableLengthBinary` traits. Also implement `Borrow`, `AsRef`, and `AsMut` for `ArrayBinary<N>` => `&Binary`.

### Why
The traits `FixedLengthBinary` and `VariableLengthBinary` add complexity to the concepts a developer needs to learn when interacting with binary that is not outsized by the benefits the type add.

The types make it possible for functions to indicate if when accepting a mutable reference to a binary if that function is going to modify the binary in a way that will change its size or not. In conversations with @jonjove he mentioned he has only found a use for this benefit in tests and not really in contract code.

I think we can support most use cases with the `Binary` and `ArrayBinary` types, without the traits.

For any function wanting to accept either a `Binary` or `ArrayBinary` the traits are unnecessary, because the function can accept either by accepting `Borrow<Binary>`, or `AsRef<Binary>`.

For #173

### CODE REVIEW

File: sdk/src/binary.rs [hunk 1]
```diff
@@ -371,35 +350,45 @@ impl<const N: u32> Ord for ArrayBinary<N> {
...
+impl<const N: u32> Borrow<Binary> for &mut ArrayBinary<N> {
+    fn borrow(&self) -> &Binary {
+        &self.0
     }
+}
 
-    #[inline(always)]
-    fn front(&self) -> u8 {
-        self.0.front()
+impl<const N: u32> BorrowMut<Binary> for &mut ArrayBinary<N> {
```
Comment (jonjove) [inline sdk/src/binary.rs:377]: Isn't this unsafe? I could change the length of an `ArrayBinary` with this.

File: sdk/src/binary.rs [hunk 2]
```diff
@@ -371,35 +350,45 @@ impl<const N: u32> Ord for ArrayBinary<N> {
...
-    #[inline(always)]
-    fn back(&self) -> u8 {
-        self.0.back()
+impl<const N: u32> AsRef<Binary> for ArrayBinary<N> {
+    fn as_ref(&self) -> &Binary {
+        &self.0
+    }
+}
+
+impl<const N: u32> AsMut<Binary> for ArrayBinary<N> {
```
Comment (jonjove) [inline sdk/src/binary.rs:389]: Isn't this unsafe? I could change the length of an `ArrayBinary` with this.

**Review by jonjove — COMMENTED** [review]
This concept seems fine, but aren't some of these conversions unsafe?

File: sdk/src/binary.rs (hunk 1 above)
Comment (leighmcculloch) [reply sdk/src/binary.rs:377]: You're right. I'll remove the BorrowMut and AsMut.

**Pull request merged**